modules/misc/nixgl.nix

```diff
@@ -0,0 +1,103 @@
+{ config, pkgs, lib, ... }:
+
+let
+  cfg = config.workarounds.nixgl;
+
+  # Expects a PackageConfiguration hashset
+  # Adapted from https://nixos.wiki/wiki/Nix_Cookbook
+  wrapWithNixGL = { binary, pkg, ... }:
+    (let
+      bin = if binary == null then pkg.pname else binary;
+      nixGL = import cfg.src { inherit pkgs; };
+      wrapperPkg = lib.attrsets.getAttrFromPath cfg.wrapperPkgPath nixGL;
+      wrapperBinPath = "${wrapperPkg}/bin/${cfg.wrapperBinName}";
+    in pkgs.runCommand "nixgl-${pkg.pname}" {
+      buildInputs = [ pkgs.makeWrapper ];
+    } ''
+      mkdir $out
+      # Link every top-level folder from pkgs.hello to our new target
+      ln -s ${pkg}/* $out
+      # Except the bin folder
+      rm $out/bin
+      mkdir $out/bin
+      # We create the bin folder ourselves and link every binary in it
+      ln -s ${pkg}/bin/* $out/bin
+      # Except the target binary
+      rm $out/bin/${bin}
+      # Because we create this ourself, by creating a wrapper
+      makeWrapper ${wrapperBinPath} $out/bin/${bin} \
+        --add-flags ${pkg}/bin/${bin}
+    '');
+  PackageConfiguration = with lib;
+    types.submodule {
+      options = {
+        binary = mkOption {
+          type = types.nullOr types.str;
+          example = "alacritty";
+          default = null;
+          description = ''
+            Name of the binary in bin/ of the package.
+            By default this uses `package.pname`.
+          '';
+        };
+        pkg = mkOption {
+          type = types.package;
+          example = "pkgs.alacritty";
+          description = ''
+            The package to be wrapped.
+            It's expected to have a `pname` attribute.
+          '';
+        };
+      };
+    };
+in {
+  meta.maintainers = with lib.maintainers; [ michaelCTS ];
+  options.workarounds.nixgl = {
+    src = lib.mkOption {
+      type = lib.types.package;
+      default = builtins.fetchTarball
+        "https://github.com/nix-community/nixGL/archive/main.tar.gz";
```

Comment:
The user will have to provide this themselves. We cannot have the situation that HM downloads a random revision of some repository. Further, this will add a documentation dependency on this repository, i.e., pretty much all HM users would need to download this repository, whether or not they use nixGL.

```suggestion
```

Reply:
I was hoping lazy eval wouldn't evaluate it until a user put stuff into `.packages`, but the example should give enough guidance :+1:

Reply:
Applied and force-pushed.